employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/pages/rotation/ShiftTemplateEditForm.java

```diff
@@ -0,0 +1,262 @@
+    private ListBox employeeSelect; //TODO: Replace with widget that has search capabilities
+
+    @DataField("delete-button")
+    private HTMLButtonElement deleteButton;
+
+    @DataField("apply-button")
+    private HTMLButtonElement applyButton;
+
+    private TenantStore tenantStore;
+
+    private EventManager eventManager;
+
+    private TranslationService translationService;
+
+    private ShiftTemplateGridObject shiftTemplateGridObject;
+
+    @Inject
+    public ShiftTemplateEditForm(PopupFactory popupFactory,
+                                 HTMLDivElement root,
+                                 @Named("span") HTMLElement popupTitle,
+                                 HTMLButtonElement closeButton,
+                                 HTMLButtonElement cancelButton,
+                                 RotationTimeSelector from,
+                                 RotationTimeSelector to,
+                                 ListBox spotSelect,
+                                 ListBox employeeSelect,
+                                 HTMLButtonElement deleteButton,
+                                 HTMLButtonElement applyButton,
+                                 TenantStore tenantStore,
+                                 EventManager eventManager,
+                                 TranslationService translationService) {
+        super(popupFactory, root, popupTitle, closeButton, cancelButton);
+        this.from = from;
+        this.to = to;
+        this.spotSelect = spotSelect;
+        this.employeeSelect = employeeSelect;
+        this.deleteButton = deleteButton;
+        this.applyButton = applyButton;
+        this.tenantStore = tenantStore;
+        this.eventManager = eventManager;
+        this.translationService = translationService;
+    }
+
+    public void init(final ShiftTemplateGridObject shiftTemplateGridObject) {
+        this.shiftTemplateGridObject = shiftTemplateGridObject;
+        shiftTemplateGridObject.setSelected(true);
+        final ShiftTemplateView template = shiftTemplateGridObject.getShiftTemplateModel().getShiftTemplateView();
+        setup(template);
+        showFor(shiftTemplateGridObject);
+    }
+
+    public void createNewShiftTemplate() {
+        setup(new ShiftTemplateView(tenantStore.getCurrentTenantId(), new ShiftTemplate(tenantStore.getCurrentTenantId(), new Spot(), 0, LocalTime.MIDNIGHT, 1, LocalTime.MIDNIGHT)));
+        spotSelect.setEnabled(true);
+        deleteButton.remove();
+        setTitle(translationService.format(I18nKeys.ShiftRosterToolbar_createShift));
+        show();
+    }
+
+    private void setup(ShiftTemplateView shiftTemplateView) {
+        employeeSelect.clear();
+        employeeSelect.addItem(translationService.format(I18nKeys.Shift_unassigned), "-1");
+
+        SpotRestServiceBuilder.getSpotList(tenantStore.getCurrentTenantId(), FailureShownRestCallback.onSuccess(spots -> {
+            spots.forEach(s -> this.spotSelect.addItem(s.getName(), s.getId().toString()));
```

Comment:
The spot select box appears to never be enabled; in this case it doesn't make any sense to enumerate all the spots there - you need just one.

Opposed to ShiftEditForm, here is no "create button" that would use enabled select box.